helios.js

```diff
@@ -17108,6 +17111,277 @@ class StakingCertifyingPurposeType extends BuiltinEnumMember {
 	}
 }
 
+/**
+ * Builtin ScriptPurpose type (Minting| Spending| Rewarding | Certifying)
+ */
+ class ScriptPurposeType extends BuiltinType {
+	toString() {
+		return "ScriptPurpose";
+	}
+
+	/**
+	 * @param {Site} site 
+	 * @param {Type} type 
+	 * @returns {boolean}
+	 */
+	isBaseOf(site, type) {
+		let b = super.isBaseOf(site, type) ||
+				(new MintingScriptPurposeType()).isBaseOf(site, type) || 
+				(new SpendingScriptPurposeType()).isBaseOf(site, type) || 
+				(new RewardingScriptPurposeType()).isBaseOf(site, type) || 
+				(new CertifyingScriptPurposeType()).isBaseOf(site, type); 
+
+		return b;
+	}
+
+	/**
+	 * @param {Word} name 
+	 * @returns {EvalEntity}
+	 */
+	getTypeMember(name) {
+		switch (name.value) {
+			case "Minting":
+				return new MintingScriptPurposeType();
+			case "Spending":
+				return new SpendingScriptPurposeType();
+			case "Rewarding":
+				return new RewardingScriptPurposeType();
+			case "Certifying":
+				return new CertifyingScriptPurposeType();
+			default:
+				return super.getTypeMember(name);
+		}
+	}
+
+	/**
+	 * @param {Site} site 
+	 * @returns {number}
+	 */
+	nEnumMembers(site) {
+		return 4;
+	}
+
+	get path() {
+		return "__helios__scriptpurpose";
+	}
+}
+
+/**
+ * Builtin ScriptPurpose::Minting
+ */
+class MintingScriptPurposeType extends BuiltinEnumMember {
+	constructor() {
+		super(new ScriptPurposeType());
+	}
+
+	toString() {
+		return "ScriptPurpose::Minting";
+	}
+
+	/**
+	 * @param {Word} name 
+	 * @returns {EvalEntity}
+	 */
+	getTypeMember(name) {
+		switch (name.value) {
+			case "from_data":
+				throw name.referenceError(`'${this.toString()}::from_data' undefined`);
+			default:
+				return super.getTypeMember(name);
+		}
+	}
+
+	/**
+	 * @param {Word} name 
+	 * @returns {Instance}
+	 */
+	getInstanceMember(name) {
+		switch (name.value) {
+			case "__eq":
+			case "__neq":
+				return Instance.new(new FuncType([new ScriptPurposeType()], new BoolType()));
+			case "policy_hash":
+				return Instance.new(new MintingPolicyHashType());
+			default:
+				return super.getInstanceMember(name);
+		}
+	}
+
+	/**
+	 * @param {Site} site 
+	 * @returns {number}
+	 */
+	getConstrIndex(site) {
+		return 1;
```

Comment:
I believe constructor indices are zero based, so Minting -> `0` (not `1`), Spending -> `1`, Rewarding -> `2`, Certifying -> `3`